src/main/java/org/apache/pulsar/io/jcloud/sink/CloudStorageGenericRecordSink.java

```diff
@@ -100,14 +103,16 @@ protected BlobStoreContext buildBlobStoreContext(CloudStorageSinkConfig sinkConf
         overrides.setProperty(Constants.PROPERTY_SO_TIMEOUT, "25000");
         overrides.setProperty(Constants.PROPERTY_MAX_RETRIES, Integer.toString(100));
 
+        if (sinkConfig.getProvider().equalsIgnoreCase(PROVIDER_GCS)) {
+            ProviderRegistry.registerProvider(new GoogleCloudStorageProviderMetadata());
+        }
+
         ContextBuilder contextBuilder = ContextBuilder.newBuilder(sinkConfig.getProvider())
                 .credentialsSupplier(getCredentialsSupplier(sinkConfig));
 
         if (StringUtils.isNotEmpty(sinkConfig.getEndpoint())) {
             contextBuilder.endpoint(sinkConfig.getEndpoint());
-            if (sinkConfig.getProvider().equalsIgnoreCase(PROVIDER_AWSS3)) {
-                overrides.setProperty(S3Constants.PROPERTY_S3_VIRTUAL_HOST_BUCKETS, "false");
-            }
+            overrides.setProperty(S3Constants.PROPERTY_S3_VIRTUAL_HOST_BUCKETS, "false");
```

Comment:
what happens to `aws-s3` case? Should we set `false` also here?

Reply:
i removed the check is because the property is only for s3 provider, and will not been used for other providers, but it appears to be a bad move, since the check can prevent non related property in the overrides. I have reverted it.